content/reference/sql/access-management.md

```diff
@@ -0,0 +1,289 @@
+Currently, the only way to remotely interact with a dolt database is when a database is running under server mode, therefore it is the only context where it is sensible for users to exist.
+As all other commands require file-level access to a database's files, any possible protection offered by other commands enabling users could easily be circumvented by directly modifying the data within the database files.
+
+In addition to privilege checks only being enabled when running in server mode, any changes to the grant tables will only persist in server mode.
+If any commands that modify the grant tables are used outside the server mode, those commands will fail, as the grant tables will not exist (they're only created when in server mode).
+
+### CLI Argument
+
+To enable full user support when using [`sql-server`](../cli.md#dolt-sql-server) mode, you may pass in the `--privilege-file="PATH"` argument.
+`"PATH"` represents the path to the privileges file, generally named `privs.json`.
+
+### YAML Configuration Option
+
+To enable full user support when using [`sql-server`](../cli.md#dolt-sql-server) mode, you may add the `privilege_file: PATH` line to your [YAML config](configuration.md).
+`PATH` represents the path to the privileges file, generally named `privs.json`.
+
+## User and Password Arguments
+
+Before the introduction of users and privileges, Dolt supported only a single user with an accompanying password.
+This was done using the `--user` and `--password` arguments ([see the docs for their defaults](../cli.md#dolt-sql-server), also available using YAML configuration), whereby a server would only allow connections that supplied that singular user and password combination.
+Although Dolt now supports users in a similar fashion to MySQL, we still retain the user and password arguments.
+In MySQL, the default super account (generally called the root user) is created during installation and configuration.
+Rather than creating a parallel with [`init`](../cli.md#dolt-init), we handle the super account creation when starting a server via the arguments.
+
+This leads to an interaction with the [privilege file](#privilege-file) that should be noted.
+When the privilege file is specified, it will be created as an empty file if it does not exist.
+As soon as any statement that modifies the grant tables (`CREATE USER`, `GRANT`, `REVOKE`, etc.) executes, the users and all privileges will save to the privilege file.
+This **includes** the super account as defined by the user and password arguments, therefore it is recommended that the super account is deleted after all users are set up, or it is given a strong password.
+On subsequent server starts, if the [privilege file](#privilege-file) _contains any data_, the user and password arguments are fully ignored.
+This behavior was chosen so that server should always have at least one user that a client may log into, otherwise the server would be completely inaccessible.
```

Comment:
Great to call out this quirk. This seems like something that could really confuse customers if they hit it.

It would be good to have dolt sql-server emit a warning message if it detects that a user/pass was specified on that command line that won't be used because of existing data in the privs file.

Reply:
Fantastic idea!